flocker/node/agents/blockdevice.py

```diff
@@ -1673,10 +1673,9 @@ def is_existing_block_device(dataset_id, path):
                 applications=None,
                 used_ports=None,
             ),
+            NonManifestDatasets(datasets=nonmanifest),
         )
 
-        if nonmanifest:
-            state += (NonManifestDatasets(datasets=nonmanifest),)
```

Comment:
Hm.  I think this breaks real-world use.  Now ... why do I think that.  Well, I wrote this originally and at the time I guess I thought the `if nonmanifest:` guard was important... But that's not really a why...

:thinks about it on the way to the office: